src/lib/sql/sql_pipeline.cpp

```diff
@@ -81,17 +88,30 @@ SQLPipeline::SQLPipeline(const std::string& sql, std::shared_ptr<TransactionCont
   _requires_execution = seen_altering_statement && _num_statements > 1;
 }
 
+const std::vector<std::string>& SQLPipeline::get_sql_strings() {
+  if (!_sql_strings.empty()) {
+    return _sql_strings;
+  }
+
+  _sql_strings.reserve(_num_statements);
+  for (auto& pipeline_statement : _sql_pipeline_statements) {
+    _sql_strings.push_back(pipeline_statement->get_sql_string());
+  }
+
+  return _sql_strings;
+}
+
 const std::vector<std::shared_ptr<hsql::SQLParserResult>>& SQLPipeline::get_parsed_sql_statements() {
   if (!_parsed_sql_statements.empty()) {
     return _parsed_sql_statements;
   }
 
   _parsed_sql_statements.reserve(_num_statements);
-  for (auto& pipeline : _sql_pipeline_statements) {
+  for (auto& pipeline_statement : _sql_pipeline_statements) {
     try {
-      _parsed_sql_statements.push_back(pipeline->get_parsed_sql_statement());
+      _parsed_sql_statements.push_back(pipeline_statement->get_parsed_sql_statement());
     } catch (const std::exception& exception) {
```

Comment:
@why do we need all these try/catch blocks here? I thought we used exceptions only in cases that we cannot recover from.

Reply:
We can't recover from this error for the actual query, but we wan't to pass on information as to where the error occurred.

Reply:
Okay. I never know when it’s best to throw an exception or return some kind of error flag. String parsing sounds like it could be something best handled by an exception. @mrks, do we have a general guideline about this?

Reply:
Do we have an official guideline now?